src/Mod/Sketcher/Gui/DrawSketchHandler.cpp

```diff
@@ -600,19 +604,19 @@ void DrawSketchHandler::createAutoConstraints(const std::vector<AutoConstraint>
                                      , geoId1, posId1, geoId2);
                 } break;
             case Sketcher::Horizontal: {
-                Gui::cmdAppObjectArgs(sketchgui->getObject(), "addConstraint(Sketcher.Constraint('Horizontal',%i)) ", geoId1);
+                Gui::cmdAppObjectArgs(sketchgui->getObject(), "addConstraint(Sketcher.Constraint('Horizontal',%i)) ",
+                                      geoId2 != Constraint::GeoUndef ? geoId2 : geoId1);

```

Comment:
I find this geoId selection behind the scenes unexpected for a user. My rationale goes like this: the function (createAutoConstraint) takes a geoId1. For a horizontal constraint on an edge there is only one geoId. A user of the function would, I think, expect that the autoconstraint is applied on geoId1. I do understand that this is not the wanted functionality in this example. geoId2, is used for the other cases as an additional geoId for constraints needing two geoIds. If after giving it a thought, you still think this is the best way, I would appreciate if you could add a minimalistic documentation to the function indicating that if the GeoId stored in the Autoconstraint is defined, it takes precedence over the geoId passed as a parameter.

Reply:
@abdullahtahiriyo I added a comment that hopefully may help.
Honestly, I perfectly understood what geoId2 was made for. Actually you can see in the commit message that I said all this is a "hack". :)
I thought about many other "cleaner" ways, even starting some coding. But I found each to require far bigger codebase changes, and being very fat for what it was intended. Finally I found this hack was probably a good solution and should serve its purpose gently. So I pushed the PR.
But definitely don't merge it if you're not totally fine with it. ;)